src/toolchain.rs

```diff
@@ -371,6 +376,42 @@ impl<'a> Toolchain<'a> {
         Ok(None)
     }
 
+    fn maybe_do_rust_analyzer_fallback(&self, binary: &str) -> anyhow::Result<Option<Command>> {
+        if binary != "rust-analyzer" && binary != "rust-analyzer.exe"
+            || self.binary_file("rust-analyzer").exists()
+        {
+            return Ok(None);
+        }
+
+        let proc = self.cfg.process;
+        let Some(path) = proc.var_os("PATH") else {
+            return Ok(None);
+        };
+
+        let me = env::current_exe()?;
+
+        // Try to find the first `rust-analyzer` under the `$PATH` that is both
+        // an existing file and not the same file as `me`, i.e. not a rustup proxy.
+        Ok(env::split_paths(&path).find_map(|mut p| {
```

Comment:
Given that there's a pretty long closure being passed to `find_map()` and not much else is happening here, suggest writing a `for`-loop instead of an iterator combinator.

Reply:
I personally think the readability is about the same either way, but I've fixed it anyway.